master/internal/api_trials_intg_test.go

```diff
@@ -652,13 +655,17 @@ func TestUnusualMetricNames(t *testing.T) {
 	req := &apiv1.CompareTrialsRequest{
 		TrialIds:      []int32{int32(trial.ID)},
 		MaxDatapoints: 3,
-		MetricNames:   []string{"a.loss", "b/loss", asciiSweep},
+		MetricNames:   []string{"a.loss", "b/loss", asciiSweep, longMetric},
 		StartBatches:  0,
 		EndBatches:    1000,
 		MetricType:    apiv1.MetricType_METRIC_TYPE_VALIDATION,
 	}
-	_, err = api.CompareTrials(ctx, req)
+	resp, err := api.CompareTrials(ctx, req)
 	require.NoError(t, err)
+
+	valuesMap := resp.Trials[0].Metrics[0].Data[0].Values.AsMap()
+	require.Len(t, valuesMap, len(expectedMetricsMap))
+	require.Equal(t, valuesMap[longMetric], expectedMetricsMap[longMetric])
```

Comment:
to check if I'm understading this correctly would this still pass if the api server restarts?

Reply:
how is this related to server restarts? it's just checking if you can write and read back the metrics with "unusual" names.

Reply:
I was confused since thought we might have changed how we store metric names (as in metrics somehow persisting in a db col as with a newer schema change) and thus the mapping you added (`type safeMetricToColumnMap struct {`) not being persisted was weird. I looked at the db queries and I understand it makes sense!

before
```sql
SELECT (select setseed(1)) as _seed, total_batches as batches, trial_id, end_time as time, (metrics->'avg_metrics'->>'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')::float8 as "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", (metrics->'avg_metrics'->>'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')::float8 as "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", (metrics->'avg_metrics'->>'loss')::float8 as "loss", (metrics->'avg_metrics'->>'epoch')::text as "epoch" FROM "metrics" WHERE (partition_type = 'TRAINING') AND (archived = FALSE) AND (trial_id = 26) AND (total_batches >= 0) AND (total_batches <= 0 OR total_batches <= 2147483647) AND (end_time > '0001-01-01 00:00:00+00:00') ORDER BY random() LIMIT 1500
```
after:

```sql
SELECT (select setseed(1)) as _seed, total_batches as batches, trial_id, end_time as time, (metrics->'avg_metrics'->>'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')::float8 as "aaaaaaaaaaaaaaaaaaaaaaaaaaa9ac16b46-b25f-4b30-85f4-88365e516ab7", (metrics->'avg_metrics'->>'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')::float8 as "aaaaaaaaaaaaaaaaaaaaaaaaaaac22fa495-57b9-4378-b186-4afe3d8e42dc", (metrics->'avg_metrics'->>'loss')::float8 as "loss", (metrics->'avg_metrics'->>'epoch')::text as "epoch" FROM "metrics" WHERE (partition_type = 'TRAINING') AND (archived = FALSE) AND (trial_id = 26) AND (total_batches >= 0) AND (total_batches <= 0 OR total_batches <= 2147483647) AND (end_time > '0001-01-01 00:00:00+00:00') ORDER BY random() LIMIT 1500
```



  | _seed | batches | trial_id | time                          | aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa | aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa | loss               | epoch  |

Reply:
ah yes; this object is local to the `MetricsTimeSeries` call.